templates/fine-tune-llm_v2/cookbooks/continue_from_checkpoint/README.ipynb

```diff
@@ -0,0 +1,118 @@
+    "\n",
+    "## How to fine-tune from a previous checkpointing"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# [Optional] You can set the WandB API key to track model performance\n",
+    "# !export WANDB_API_KEY={YOUR_WANDB_API_KEY}\n",
+    "\n",
+    "# Continue LoRA fine-tuning on the GSM8k dataset with Llama 3 8B\n",
+    "!llmforge dev finetune llama-3-8b.yaml"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## What we are fine-tuning on\n",
+    "\n",
+    "Running the above command will fine-tune on the [GSM8k dataset](https://huggingface.co/datasets/gsm8k). \n",
+    "In this example, we split the dataset into two halfs, each consisting of approximately 4.000 samples.\n",
+    "The provided initial checkpoint has been trained on the first half and is already good at solving GSM8k. By running the above command, you continue fine-tuning from the provided checkpoint with the second half.\n",
+    "\n",
+    "Note the following evaluation losses. The first graph shows the evaluation loss on three epochs of training on the first half of the GSM8k dataset.\n",
+    "\n",
+    "<img src=\"../../assets/3epochs_1st_dataset.png\" alt=\"evaluation loss of 1st training\" width=\"700\"/>\n",
+    "\n",
+    "The second graph shows the evaluation loss on three epochs of training on the second half, starting with the fine-tuned weights of the first training.\n",
+    "\n",
+    "<img src=\"../../assets/3epochs_2nd_dataset.png\" alt=\"evaluation loss of 2nd training\" width=\"700\"/>\n",
+    "\n",
+    "Note that the evaluation loss starts way lower than where it starts or finishes in the first training.\n",
+    "\n",
+    "## What and how are we fine-tuning?\n",
+    "\n",
+    "The following is a snippet from the `llama-3-8b.yaml` file we use above. \n",
+    "\n",
+    "```yaml\n",
+    "# ...\n",
+    "model_id: meta-llama/Meta-Llama-3-8B-Instruct\n",
+    "# initial_base_model_ckpt_path: ...\n",
+    "initial_adapter_model_ckpt_path: s3://large-dl-models-mirror/finetuning_template/continued_ft_gsm8k_checkpoint\n",
+    "train_path: s3://large-dl-models-mirror/finetuning_template/train_2.jsonl\n",
+    "# ...\n",
+    "```\n",
+    "\n",
+    "We fine-tune Llama 3 8B Instruct, but the initial weights of the LoRA adapter are loaded from our s3 mirror.\n",
+    "It makes sense to keep those weights in a bucket so that they can be accessed from all nodes of your cluster.\n",
+    "The train path `(.../train_2.jsonl)` points to the second part of the GSM8k dataset that we fine-tune on.\n",
+    "If we were not fine-tuning with LoRA, we would not configure `initial_adapter_model_ckpt_path`, but `initial_base_model_ckpt_path` instead.\n",
```

Comment:
```suggestion
    "If we wanted to continue the finetuning of a full-parameter checkpoint, we should configure `initial_base_model_ckpt_path` instead of `initial_adapter_model_ckpt_path`. \n",
```